xuluomin/src/server/src/server.cpp

```diff
@@ -0,0 +1,79 @@
+#include <ros/ros.h>
+#include <time.h>
+
+#include <client/time.h>
+#include <client/show.h>
+
+#include <vector>
+#include <iostream>
+
+client::show srv;
+
+std::vector<ros::Subscriber> server_subscribers;
+
+ros::ServiceServer service_server;
+
+void time_callback(const client::time::ConstPtr& time)
+{
+    time_t  cur_t=time->second;
+    struct tm *t;
+    t=localtime(&cur_t);
+
+    int year = t->tm_year + 1900; // years since 1900
+    int month = t->tm_mon + 1; // months since January - [0, 11]
+    int day = t->tm_mday;
+
+    int hour = t->tm_hour;
+    int minute = t->tm_min;
+    int second = t->tm_sec;
+
+    ROS_INFO("The name of client:%s   time: %d-%d-%d  %d : %d : %d  ",
+             time->name.c_str(), year, month, day, hour, minute, second);
+}
+
+// service回调函数，输入参数request，输出参数response
+bool show_callback(client::show::Request &request,
+                   client::show::Response &response)
+{
+    if(request.request == 1)
+    {
+        ROS_INFO("The name of client: %s logged in",
+                 request.name.c_str());
+        response.response = 10;
+        server_subscribers.push_back(n_pointer->subscribe(request.name, 10, time_callback));
+    }
+    else
+    {
+        ROS_INFO("The name of client: %s logged out",
+                 request.name.c_str());
+        for(auto &subscriber : server_subscribers)
+        {
+            std::string the_string = "/";
+            if(the_string + request.name.c_str() == subscriber.getTopic())
+                subscriber.shutdown();
+        }
+
+        server_subscribers.clear();
+        response.response = 20;
+
```

Comment:
**[cpplint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br> Redundant blank line at the end of a code block should be deleted.  [whitespace/blank_line] [3]